File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/psychotypeDescription/DescriptionViewModel.java
package com.laushkina.anastasia.psychosophy.view.psychotypeDescription;

import android.graphics.drawable.Drawable;

public class DescriptionViewModel {
    private CharSequence firstFunctionDescription;
    private CharSequence secondFunctionDescription;
    private CharSequence thirdFunctionDescription;
    private CharSequence forthFunctionDescription;
    private CharSequence shortDescription;
    private Drawable image;

    private DescriptionViewModel(Builder builder) {
        firstFunctionDescription = builder.firstFunctionDescription;
        secondFunctionDescription = builder.secondFunctionDescription;
        thirdFunctionDescription = builder.thirdFunctionDescription;
        forthFunctionDescription = builder.forthFunctionDescription;
        shortDescription = builder.shortDescription;
        image = builder.image;
    }

    public CharSequence getFirstFunctionDescription() {
        return firstFunctionDescription;
    }

    public CharSequence getSecondFunctionDescription() {
        return secondFunctionDescription;
    }

    public CharSequence getThirdFunctionDescription() {
        return thirdFunctionDescription;
    }

    public CharSequence getForthFunctionDescription() {
        return forthFunctionDescription;
    }

    public CharSequence getShortDescription() {
        return shortDescription;
    }

    public Drawable getImage() {
        return image;
    }

    public static class Builder {
        private CharSequence firstFunctionDescription;
        private CharSequence secondFunctionDescription;
        private CharSequence thirdFunctionDescription;
        private CharSequence forthFunctionDescription;
        private CharSequence shortDescription;
        private Drawable image;

        public Builder() {}

        public Builder firstFunctionDescription(CharSequence firstFunctionDescription) {
            this.firstFunctionDescription = firstFunctionDescription;
            return this;
        }

        public Builder secondFunctionDescription(CharSequence secondFunctionDescription) {
            this.secondFunctionDescription = secondFunctionDescription;
            return this;
        }

        public Builder thirdFunctionDescription(CharSequence thirdFunctionDescription) {
            this.thirdFunctionDescription = thirdFunctionDescription;
            return this;
        }

        public Builder forthFunctionDescription(CharSequence forthFunctionDescription) {
            this.forthFunctionDescription = forthFunctionDescription;
            return this;
        }

        public Builder shortDescription(CharSequence shortDescription) {
            this.shortDescription = shortDescription;
            return this;
        }

        public Builder image(Drawable image) {
            this.image = image;
            return this;
        }

        public DescriptionViewModel build() {
            return new DescriptionViewModel(this);
        }
    }
}

File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/psychotypeDescription/DescriptionViewModel.kt
package com.laushkina.anastasia.psychosophy.view.psychotypeDescription

import android.graphics.drawable.Drawable

class DescriptionViewModel(val firstFunctionDescription: CharSequence?,
                           val secondFunctionDescription: CharSequence?,
                           val thirdFunctionDescription: CharSequence?,
                           val forthFunctionDescription: CharSequence?,
                           val shortDescription: CharSequence?,
                           val image: Drawable?) {

    constructor(builder: Builder) : this(
            builder.firstFunctionDescription,
            builder.secondFunctionDescription,
            builder.thirdFunctionDescription,
            builder.forthFunctionDescription,
            builder.shortDescription,
            builder.image
    )


    class Builder {
        var firstFunctionDescription: CharSequence? = null
        var secondFunctionDescription: CharSequence? = null
        var thirdFunctionDescription: CharSequence? = null
        var forthFunctionDescription: CharSequence? = null
        var shortDescription: CharSequence? = null
        var image: Drawable? = null

        fun firstFunctionDescription(firstFunctionDescription: CharSequence): Builder {
            this.firstFunctionDescription = firstFunctionDescription
            return this
        }

        fun secondFunctionDescription(secondFunctionDescription: CharSequence): Builder {
            this.secondFunctionDescription = secondFunctionDescription
            return this
        }

        fun thirdFunctionDescription(thirdFunctionDescription: CharSequence): Builder {
            this.thirdFunctionDescription = thirdFunctionDescription
            return this
        }

        fun forthFunctionDescription(forthFunctionDescription: CharSequence): Builder {
            this.forthFunctionDescription = forthFunctionDescription
            return this
        }

        fun shortDescription(shortDescription: CharSequence): Builder {
            this.shortDescription = shortDescription
            return this
        }

        fun image(image: Drawable): Builder {
            this.image = image
            return this
        }

        fun build(): DescriptionViewModel {
            return DescriptionViewModel(this)
        }
    }
}